src/python/__init__.py

```diff
@@ -326,3 +330,36 @@ def decompress_file(src, dst, comtype, stat=None):
 compression_suffix  = _createrepo_c.compression_suffix
 detect_compression  = _createrepo_c.detect_compression
 compression_type    = _createrepo_c.compression_type
+
+
+DATA = os.path.join(os.path.dirname(__file__), 'data')
+
+# Support running tests from the source tree
+if not os.path.exists(DATA):
+    from skbuild.constants import CMAKE_INSTALL_DIR as SKBUILD_CMAKE_INSTALL_DIR
```

Comment:
We can't rely on this - most users won't be installing via PyPI, they'll be using the normal RPM.  And even if they're installing from PyPI, hopefully we'll be able to provide a bdist eventually, so there's no need to have a runtime dep on ```scikit-build```.

This causes an import error when following the normal build / test instructions in the README.

Putting that aside, running tests using the normal instructions has been broken by a change somewhere.  I get 

```
ERROR: Failure: ImportError (cannot import name _createrepo_c)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/vagrant/devel/createrepo_c/tests/python/tests/test_xml_parser.py", line 6, in <module>
    import createrepo_c as cr
  File "/home/vagrant/devel/createrepo_c/build/src/python/createrepo_c/__init__.py", line 8, in <module>
    from . import _createrepo_c
ImportError: cannot import name _createrepo_c
```

From running ``` PYTHONPATH=`readlink -f ./build/src/python/` nosetests -s tests/python/tests/````

despite that the path does have ```_createrepo_c.so``` on it.

```
[vagrant@pulp3 createrepo_c]$ ls build/src/python/createrepo_c/
_createrepo_c.so  __init__.py  __init__.pyc
```

Looking into what might be causing this currently.

Reply:
>We can't rely on this - most users won't be installing via PyPI

This was only to support the case of local installation  where the developer would create an editable install using `python install -e .` or `python setup.py develop`

> running tests using the normal build process documented in the README has been broken by a change somewhere

More details in this post: https://groups.google.com/d/msg/scikit-build/XYfZFT2iFJM/lFEaEnm3CQAJ

Reply:
To summarize, there are two issues preventing editable from working.

By editable install, I mean the ability to run `pip install -e .` and then import `createrepo_c`

(1) python source should be moved from `src/python/*` to `src/python/createrepo_c/*`

(2) resolving the location of the compiled module should be fixed by either having a fallback to `skbuild.constant` or by hardcoding the expected location

Reply:
@dralley The python bindings are shipped in Fedora/CentOS, Mageia, and openSUSE. Only SUSE Linux Enterprise currently does not ship them.

Reply:
@Conan-Kudo Ah, I hadn't noticed it was built as a separate RPM package (```python3-createrepo_c```).  Thanks.

Reply:
@jcfr I've moved ```src/python/__init__.py``` to ```src/python/createrepo_c/__init__.py```, so I think everything is fine now on that front.  My editable installation worked.

re: 2, Here is a possible solution (incomplete commit, going to lunch)

https://github.com/rpm-software-management/createrepo_c/pull/103/commits/58637bf102d6345f83702e8c72297bb2db66a339